### PR TITLE
Orthographic camera support

### PR DESCRIPTION
**Pull request by Guo-Haowei**
* Add `es2015.core` lib in `tsconfig.josn` in order to use `Math.sign()`

### CODE REVIEW

File: src/examples/example.ts [hunk 1]
```diff
@@ -4,7 +4,8 @@ import ComboControls from '../';
 const width = window.innerWidth;
 const height = window.innerHeight;
 const scene = new THREE.Scene();
-const camera = new THREE.PerspectiveCamera(60, width / height, 0.01, 100);
+// const camera = new THREE.PerspectiveCamera(60, width / height, 0.01, 100);
```
Comment (f1cognite) [inline src/examples/example.ts]: Could you use this package https://www.npmjs.com/package/query-string in `example.ts` to be able to set a query parameter (`?orthographic`) to the URL to switch to orthograpic camera?

File: tsconfig.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
     "noImplicitAny": true,
     "strictNullChecks": true,
     "sourceMap": true,
-    "moduleResolution": "node"
+    "moduleResolution": "node",
+    "lib": ["es5", "dom", "es2015.core"]
```
Comment (f1cognite) [inline tsconfig.json]: Do you remember why we need to add this?

File: src/ComboControls.ts [hunk 3]
```diff
@@ -528,24 +536,37 @@ export default class ComboControls extends EventDispatcher {
...
       reusableVector3,
       sphericalEnd,
       targetEnd,
+      camera,
+      reusableCamera,
     } = this;
-    const { fov } = this.camera;
-    const distFromCameraToScreenCenter = Math.tan(ThreeMath.degToRad(90 - fov * 0.5));
+
+    // @ts-ignore
```
Comment (f1cognite) [inline src/ComboControls.ts]: Do you know why we need ignore here?

File: src/ComboControls.ts [hunk 4]
```diff
@@ -528,24 +536,37 @@ export default class ComboControls extends EventDispatcher {
...
+    }
 
     // we actually don't use screenWidth, since perspective camera is fixed to screen height
     this.panLeft((2 * deltaX * targetDistance) / domElement.clientHeight);
     this.panUp((2 * deltaY * targetDistance) / domElement.clientHeight);
   }
 
-  private dolly = (x: number, y: number, deltaDistance: number) => {
+  private dollyOrthographicCamera = (x: number, y: number, deltaDistance: number) => {
+    // @ts-ignore
```
Comment (f1cognite) [inline src/ComboControls.ts]: We don't need the ignore here? Use Typescript casting

File: src/ComboControls.ts [hunk 5]
```diff
@@ -71,11 +73,13 @@ export default class ComboControls extends EventDispatcher {
...
   public dispose: () => void;
+  public minZoom: number = 0;
+  public maxZoom: number = Infinity;
 
   private temporarilyDisableDamping: Boolean = false;
-  private camera: PerspectiveCamera;
+  private camera: PerspectiveCamera | OrthographicCamera;
   private firstPersonMode: Boolean = false;
-  private reusableCamera: PerspectiveCamera = new PerspectiveCamera();
+  private reusableCamera: any;
```
Comment (f1cognite) [inline src/ComboControls.ts]: We should avoid `any`

File: tsconfig.json (hunk 2 above)
Comment (Guo-Haowei) [reply tsconfig.json]: In order to use `Math.sign()`

File: src/ComboControls.ts [hunk 6]
```diff
@@ -254,7 +259,10 @@ export default class ComboControls extends EventDispatcher {
     y = (y / domElement.clientHeight) * -2 + 1;
 
     const dollyIn = delta < 0;
-    this.dolly(x, y, this.getDollyDeltaDistance(dollyIn, Math.abs(delta)));
+    const deltaDistance = this.camera instanceof PerspectiveCamera ?
+      this.getDollyDeltaDistance(dollyIn, Math.abs(delta)) :
+      Math.sign(delta) * 0.3;
```
Comment (f1cognite) [inline src/ComboControls.ts]: Why `0.3`?

File: src/ComboControls.ts [hunk 7]
```diff
@@ -508,6 +516,7 @@ export default class ComboControls extends EventDispatcher {
 
   private rotateFirstPersonMode = (azimuthAngle: number, polarAngle: number) => {
     const { camera, firstPersonRotationFactor, reusableCamera, reusableVector3, sphericalEnd, targetEnd } = this;
+    // @ts-ignore
```
Comment (f1cognite) [inline src/ComboControls.ts]: Why ignore?

File: src/examples/example.ts [hunk 8]
```diff
@@ -1,10 +1,20 @@
 import * as THREE from 'three';
 import ComboControls from '../';
+declare var require: any;
```
Comment (f1cognite) [inline src/examples/example.ts]: This shouldn't be necessary or am I mistaken?

File: src/examples/example.ts [hunk 9]
```diff
@@ -1,10 +1,20 @@
...
+const queryString = require('query-string');
 
 const width = window.innerWidth;
 const height = window.innerHeight;
 const scene = new THREE.Scene();
-const camera = new THREE.PerspectiveCamera(60, width / height, 0.01, 100);
+
+const wantOrthographic = queryString.parse(location.search).orthographic !== undefined;
```
Comment (f1cognite) [inline src/examples/example.ts]: Very good 👍

File: src/examples/example.ts (hunk 8 above)
Comment (Guo-Haowei) [reply src/examples/example.ts]: require is not defined in typescript.
https://stackoverflow.com/questions/31173738/typescript-getting-error-ts2304-cannot-find-name-require

File: src/ComboControls.ts (hunk 6 above)
Comment (Guo-Haowei) [reply src/ComboControls.ts]: It's a "magic number" I tried that work the best

File: src/examples/example.ts (hunk 8 above)
Comment (f1cognite) [reply src/examples/example.ts]: Could you rather do the TypeScript 2.x solution?

File: src/ComboControls.ts (hunk 6 above)
Comment (f1cognite) [reply src/ComboControls.ts]: We should not have any magic numbers. We should expose it as config

**Pull request merged**